kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManager.java

```diff
@@ -26,12 +27,42 @@
   /**
    * Returns the {@link RegisteredSchema schema} registered with the given {@code schemaId}.
    */
-  RegisteredSchema getSchemaById(EmbeddedFormat format, int schemaId);
+  RegisteredSchema getSchemaById(String subject, int schemaId);
 
   /**
-   * Parses and returns the potentially newly registered {@link RegisteredSchema schema}
-   * corresponding to {@code rawSchema}.
+   * Returns the {@link RegisteredSchema schema} registered with the given {@code schemaId}.
+   */
+  RegisteredSchema getSchemaById(
+      SubjectNameStrategy subjectNameStrategy, String topicName, boolean isKey, int schemaId);
```

Comment:
why the subjectNameStrategy, topicName and isKey parameters on a method names this?

Reply:
Completely rewrote this bit. Ack.